netbox/dcim/models.py

```diff
@@ -3027,15 +3027,14 @@ class Meta:
             ('termination_b_type', 'termination_b_id'),
         )
 
-    def __str__(self):
-        if self.label:
-            return self.label
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        # Save a copy of the PK on the instance since it's nullified if .delete() is called
-        if not hasattr(self, 'id_string'):
-            self.id_string = '#{}'.format(self.pk)
+        # A copy of the PK to be used by __str__ in case the object is deleted
+        self._pk = self.pk
 
-        return self.id_string
+    def __str__(self):
+        return self.label or '#{}'.format(self._pk)
```

Comment:
```suggestion
        return self.label or '#{}'.format(self.pk or self._pk)
```

Sorry, my earlier suggestion ignored the Cable creation workflow. This checks for `self.pk` first and falls back to `self._pk` only if the instance has been deleted.

Reply:
I could change that if you'd like, but it doesn't help as` __init__` will always be called in order to instantiate `self`.

If you're referring to the test, it will fail with or without the suggested change. It's the fact that the cable object is not fetched after being saved and is only getting a PK afterwards, init is not being called again so `_pk` is not being updated, and `str` is being called after the cable is being deleted, so `pk` is already nulled.

Reply:
Consider this:

```python
c = Cable(...)
c.save()
```

`c._pk` would be set to `None` when init is called on line 1, since the object hasn't been created in the database yet. `c.pk` is assigned during `save()`, but `self._pk` won't get updated on the instance.

Reply:
Yeah, so my solution was to add this during `save` (ignore the force-push; I messed up while copying)
```
self._pk = self.pk
```